src/pybind/cephfs/cephfs.pyx

```diff
@@ -1109,27 +1116,47 @@ cdef class LibCephFS(object):
         cdef:
             char* _path = path
             statx stx
+            int _flags
+
+        if basic:
+            _flags = CEPH_STATX_BASIC_STATS
+        else:
+            _flags = CEPH_STATX_ALL_STATS
 
         if follow_symlink:
             with nogil:
                 ret = ceph_statx(self.cluster, _path, &stx,
-                                 CEPH_STATX_BASIC_STATS, 0)
+                                 _flags, 0)
         else:
             with nogil:
                 ret = ceph_statx(self.cluster, _path, &stx,
-                                 CEPH_STATX_BASIC_STATS, AT_SYMLINK_NOFOLLOW)
+                                 _flags, AT_SYMLINK_NOFOLLOW)
 
         if ret < 0:
             raise make_ex(ret, "error in stat: {}".format(path.decode('utf-8')))
-        return StatResult(st_dev=stx.stx_dev, st_ino=stx.stx_ino,
-                          st_mode=stx.stx_mode, st_nlink=stx.stx_nlink,
-                          st_uid=stx.stx_uid, st_gid=stx.stx_gid,
-                          st_rdev=stx.stx_rdev, st_size=stx.stx_size,
-                          st_blksize=stx.stx_blksize,
-                          st_blocks=stx.stx_blocks,
-                          st_atime=datetime.fromtimestamp(stx.stx_atime.tv_sec),
-                          st_mtime=datetime.fromtimestamp(stx.stx_mtime.tv_sec),
-                          st_ctime=datetime.fromtimestamp(stx.stx_ctime.tv_sec))
+
+        if basic:
+            return StatResult(st_dev=stx.stx_dev, st_ino=stx.stx_ino,
+                              st_mode=stx.stx_mode, st_nlink=stx.stx_nlink,
+                              st_uid=stx.stx_uid, st_gid=stx.stx_gid,
+                              st_rdev=stx.stx_rdev, st_size=stx.stx_size,
+                              st_blksize=stx.stx_blksize,
+                              st_blocks=stx.stx_blocks,
+                              st_atime=datetime.fromtimestamp(stx.stx_atime.tv_sec),
+                              st_mtime=datetime.fromtimestamp(stx.stx_mtime.tv_sec),
+                              st_ctime=datetime.fromtimestamp(stx.stx_ctime.tv_sec))
+        else:
+            return StatResultAll(st_dev=stx.stx_dev, st_ino=stx.stx_ino,
+                              st_mode=stx.stx_mode, st_nlink=stx.stx_nlink,
+                              st_uid=stx.stx_uid, st_gid=stx.stx_gid,
+                              st_rdev=stx.stx_rdev, st_size=stx.stx_size,
+                              st_blksize=stx.stx_blksize,
+                              st_blocks=stx.stx_blocks,
+                              st_atime=datetime.fromtimestamp(stx.stx_atime.tv_sec),
+                              st_mtime=datetime.fromtimestamp(stx.stx_mtime.tv_sec),
+                              st_ctime=datetime.fromtimestamp(stx.stx_ctime.tv_sec),
+                              st_btime=datetime.fromtimestamp(stx.stx_btime.tv_sec),
+                              st_version=stx.stx_version)
```

Comment:
I think you could collapse this down to generating a StatResult and only filling out the st_btime and st_version if basic is false.

When I get a StatResult, how do I know whether I can trust the st_version and st_btime fields? I think you need some way to tell whether those fields are valid. Are they set to "None" when you don't fill them out? That might be enough, but a user of this API will need to realize to check for that unless they know a priori that those fields will be valid.